### PR TITLE
[WIP] porting diag_second_moments_ubycond function

### PR DESCRIPTION
**Pull request by xyuan**
1): porting diag_second_moments_ubycond function
2): fix a bug in previous diag_second_moment_srf c++ function, and extend the test using shcol=128, instead of 1.

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "shoc_diag_second_moments_ubycond_impl.hpp"
+#include "ekat/scream_types.hpp"
+
+namespace scream {
+namespace shoc {
+
+/*
+ *  * Explicit instantiation for doing ice melting on Reals using the
```
Comment (jeff-cohere) [inline components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond.cpp:8]: Can you get rid of the "ice melting" comment here? Let's keep this generic.

File: components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+::shoc_diag_second_moments_ubycond(
+    const MemberType& team, const Int& shcol, const Int& num_tracer,
+    const view_1d<Spack>& thl_sec, const view_1d<Spack>& qw_sec, const view_1d<Spack>& wthl_sec, const view_1d<Spack>& wqw_sec,
+    const view_1d<Spack>& qwthl_sec, const view_1d<Spack>& uw_sec, const view_1d<Spack>& vw_sec,const view_1d<Spack>& wtke_sec, 
+    const view_2d<Spack>& wtracer_sec)
+{
+  // Purpose of this subroutine is to diagnose the upper
+  //  boundary condition for the second order moments
+  //  needed for the SHOC parameterization.  Currently
+  //  set all to zero.
```
Comment (jeff-cohere) [inline components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond_impl.hpp:23]: Is this the current implementation for the boundary condition of the 2nd-order moments, or just a placeholder?

File: components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond.cpp (hunk 1 above)
Comment (xyuan) [reply components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond.cpp]: I didn't realize this one, modification is done! please check it.

File: components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond_impl.hpp (hunk 2 above)
Comment (xyuan) [reply components/scream/src/physics/shoc/shoc_diag_second_moments_ubycond_impl.hpp:23]: It sounds to me this is just placeholder

**Comment by jeff-cohere** [comment]
@xyuan, Andrew Bradley has pointed out some threading issues with #501, and possibly with this PR. I think we're going to try to cover these in Monday's telecon. Let's avoid merging this PR for the moment.

**Comment by jgfouca** [comment]
@xyuan , please rebase this on latest master

**Comment by xyuan** [comment]
Yes, I was wondering about this and ask Jim to review, but he didn’t response me. I take out the automerge. I will redo it in the way Jim’s PR

**Comment by xyuan** [comment]
I will redo this PR, so I close it.

**Pull request closed without merging**